src/mcp/mcpConnectionManager.ts

```diff
@@ -114,4 +118,30 @@ To connect, choose a connection from MongoDB VSCode extensions's sidepanel - htt
 
     await this.connectToVSCodeConnection(connectParams);
   }
+
+  overrideAppNameIfContainsVSCode(
+    connectParams: MCPConnectParams,
+  ): MCPConnectParams {
+    const connectionURL = new ConnectionString(connectParams.connectionString);
+    const connectOptions: DevtoolsConnectOptions = {
+      ...connectParams.connectOptions,
+    };
+    const searchParamAppName = connectionURL.searchParams.get('appName');
```

Comment:
```suggestion
    const searchParamAppName = connectionURL
      .typedSearchParams<DevtoolsConnectOptions>()
      .get('appName');
```

We can't type `searchParams` because it a) is a standard property with an existing type and b) because the types would differ between MongoDB driver versions, but I think it's safe to say we generally prefer type-safe access if we can